appium-tests/android/android.spec.js

```diff
@@ -0,0 +1,338 @@
+/*jshint node: true, jasmine: true */
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+*/
+
+// these tests are meant to be executed by Cordova Medic Appium runner
+// you can find it here: https://github.com/apache/cordova-medic/
+// it is not necessary to do a full CI setup to run these tests
+// just run "node cordova-medic/medic/medic.js appium --platform android --plugins cordova-plugin-contacts"
+
+'use strict';
+
+var wdHelper = require('../helpers/wdHelper');
+var screenshotHelper = require('../helpers/screenshotHelper');
+var contactsHelper = require('../helpers/contactsHelper');
+
+var STARTING_MESSAGE = 'Ready for action!';
+var MINUTE = 60 * 1000;
+var DEFAULT_WEBVIEW_CONTEXT = 'WEBVIEW';
+
+describe('Contacts Android', function () {
+    var driver;
+    // the name of webview context, it will be changed to match needed context if there are named ones:
+    var webviewContext = DEFAULT_WEBVIEW_CONTEXT;
+    // this indicates that there was a critical error and we should try to recover:
+    var errorFlag = false;
+    // this indicates that we couldn't restore Appium session and should fail fast:
+    var stopFlag = false;
+
+    function win() {
+        expect(true).toBe(true);
+    }
+
+    function fail(error) {
```

Comment:
I was not aware that they did implement it. Such a nice function, gonna use it a lot.